sdk/python/feast/infra/gcp.py

```diff
@@ -21,19 +28,37 @@ def _delete_all_values(client, key) -> None:
             client.delete(entity.key)
 
 
+def compute_datastore_entity_id(entity_key: EntityKeyProto) -> str:
+    """
+    Compute Datastore Entity id given Feast Entity Key.
+
+    Remember that Datastore Entity is a concept from the Datastore data model, that has nothing to
+    do with the Entity concept we have in Feast.
+    """
+    return mmh3.hash_bytes(serialize_entity_key(entity_key)).hex()
+
+
+def _make_tzaware(t: datetime):
+    """ We assume tz-naive datetimes are UTC """
+    if t.tzinfo is None:
+        return t.replace(tzinfo=utc)
+    else:
+        return t
+
+
 class Gcp(Provider):
-    _project_id: Optional[str]
+    _gcp_project_id: Optional[str]
 
     def __init__(self, config: Optional[DatastoreOnlineStoreConfig]):
         if config:
-            self._project_id = config.project_id
+            self._gcp_project_id = config.project_id
```

Comment:
Wait, I dont think we should be calling this `project_id` if it references a `gcp project id`? Can we differentiate those so that its less confusing?

Reply:
Also, do we infer the GCP project when the user doesn't set it? We dont want them to have to go and manually set it if we can get it from the env.

Reply:
Yea it can be omitted. I can call it `gcp_project_id` too but it goes under `datastore` section in the config so one could argue it is clear that it pertains to GCP

Reply:
wait, but why do we need to have the project defined there in any case? shouldnt we only allow one project to be used globally?

Reply:
I'm not sure how gcloud auth works exactly, but wouldn't it be useful to be able to choose a specific GCP project here?

Reply:
I'm not opposed to it, I just want to keep things as simple as possible. We already have config hell with Spark. I dont feel too strongly here though.